File: data/code_pairs/cpp/1837.cpp
// main.cpp
#include <iostream>
#include <memory>

struct t {
    std::unique_ptr<int> caf;
};

int main() {
    t x, y;
    int* ptr = nullptr;

    y.caf = std::make_unique<int>(6);  // Directly assign 6 to y.caf
    ptr = y.caf.get();

    if (y.caf == nullptr) {
        std::cerr << "Allocation failed" << std::endl;
        return 1;
    }
    if (*y.caf != 6) {
        std::cerr << "Value assignment failed" << std::endl;
        return 1;
    }

    *ptr = 123;
    if (*y.caf != 123) {
        std::cerr << "Value reassignment failed" << std::endl;
        return 1;
    }

    // Simulating the Fortran behavior by manually copying the value, not sharing ownership
    x.caf = std::make_unique<int>(*y.caf);

    *ptr = 456; // This change will not affect x.caf if they do not share the same memory
    if (*y.caf != 456) {
        std::cerr << "y.caf was not updated" << std::endl;
        return 1;
    }
    if (*x.caf == 456) { // x.caf should remain unchanged if it doesn't share memory with y.caf
        std::cerr << "Memory sharing failed" << std::endl;
        return 1;
    }

    std::cout << "All tests passed." << std::endl;
    return 0;
}

File: data/code_pairs/fortran/1837.f90
! main.f90
module m
    implicit none
    type :: t
        integer, allocatable :: CAF(:)
    end type t
end module m

program main
    use m
    type(t), target :: x, y
    integer, pointer :: ptr(:)
    integer :: i

    allocate(x%CAF(1), y%CAF(1))
    ptr => y%CAF
    ptr(1) = 6

    if (.not.allocated(x%CAF) .or. .not.allocated(y%CAF)) stop "Allocation failed"
    if (y%CAF(1) /= 6) stop "Value assignment failed"

    x%CAF = y%CAF  ! This makes a copy in Fortran, not sharing
    ptr(1) = 123

    if (y%CAF(1) /= 123) stop "Value reassignment failed"
    if (x%CAF(1) == 123) stop "Memory sharing failed"

    print *, "All tests passed."
    deallocate(x%CAF, y%CAF)
end program main